### PR TITLE
Fix Out of Memory failure in test TensorTest.Tensor64BitDimension

### PR DESCRIPTION
**Pull request by mingzhe09088**

### CODE REVIEW

**Comment by bddppq** [comment]
@caffe2bot test this please

File: caffe2/core/blob_test.cc [hunk 1]
```diff
@@ -538,7 +538,13 @@ TEST(TensorTest, Tensor64BitDimension) {
   EXPECT_EQ(tensor.ndim(), 1);
   EXPECT_EQ(tensor.dim(0), large_number);
   EXPECT_EQ(tensor.size(), large_number);
-  EXPECT_TRUE(tensor.mutable_data<char>() != nullptr);
+  try {
+    EXPECT_TRUE(tensor.mutable_data<char>() != nullptr);
```
Comment (Yangqing) [inline caffe2/core/blob_test.cc:542]: Thanks! You might want to wrap line 552 as well.

File: caffe2/core/blob_test.cc [hunk 2]
```diff
@@ -538,7 +538,13 @@ TEST(TensorTest, Tensor64BitDimension) {
...
+  } catch (const EnforceNotMet& e) {
+    string msg = e.what();
+    msg = msg.substr(0, msg.find('\n'));
+    LOG(WARNING) << msg;
```
Comment (Yangqing) [inline caffe2/core/blob_test.cc]: You may want to make sure that the error message contains "aligned_malloc" - for all others, we should still not swallow the exception.

**Review by Yangqing — CHANGES_REQUESTED** [review]

File: caffe2/core/blob_test.cc (hunk 1 above)
Comment (mingzhe09088) [reply caffe2/core/blob_test.cc:542]: Thanks! I've updated that.

File: caffe2/core/blob_test.cc (hunk 2 above)
Comment (mingzhe09088) [reply caffe2/core/blob_test.cc]: Updated as well.

**Comment by bddppq** [comment]
@caffe2bot add to whitelist

File: caffe2/core/blob_test.cc [hunk 3]
```diff
@@ -538,12 +538,19 @@ TEST(TensorTest, Tensor64BitDimension) {
...
-  // Try to go even larger, but this time we will not do mutable_data because we
-  // do not have a large enough memory.
-  tensor.Resize(large_number, 100);
+  try {
+    EXPECT_TRUE(tensor.mutable_data<char>() != nullptr);
+    // Try to go even larger, but this time we will not do mutable_data because we
+    // do not have a large enough memory.
+    tensor.Resize(large_number, 100);
+  } catch (const EnforceNotMet& e) {
+    string msg = e.what();
```
Comment (Yangqing) [inline caffe2/core/blob_test.cc]: As mentioned in the above comment: you should do a string search for aligned_malloc, and only when the exception is caused by aligned malloc should we ignore it.

**Review by Yangqing — CHANGES_REQUESTED** [review]

File: caffe2/core/blob_test.cc (hunk 1 above)
Comment (bddppq) [reply caffe2/core/blob_test.cc:542]: @Yangqing looks to me `Resize` doesn't allocate memory

**Comment by bddppq** [comment]
@mingzhe09088 We should only wrap the line(s) that throw the OOM exception in the test.

**Comment by pjh5** [comment]
I am not sure why the conda build is still failing, but it looks like this error was fixed on it. I would still check this in despite the conda failure

**Comment by mingzhe09088** [comment]
@bddppq From the previous testing result, when the OOM error happened, all the following lines in that function also failed. So I protect the following lines.

**Comment by bddppq** [comment]
@mingzhe09088 In case OOM error happened the rest of the test should be skipped (just early return).

**Comment by mingzhe09088** [comment]
@bddppq Thanks! I've pushed a fix.

**Review by Yangqing — APPROVED** [review]
Looks pretty good! :)

**Review by bddppq — APPROVED** [review]
Looks good. Thanks

**Pull request merged**